File: bench/benches/reverse.lua
-- reverse a 20000-element table each iteration; checksum = new head = 19999.
package.path = (arg[0]:match("(.*/)") or "./") .. "../lib/?.lua;" .. package.path
local bench = require("bench")
local data = {}
for i = 0, 19999 do data[i + 1] = i end
bench("reverse", function()
  local n = #data
  local r = {}
  for i = 1, n do r[i] = data[n - i + 1] end
  return r[1]
end)

File: bench/benches/reverse.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "lib"))
from bench import bench

data = list(range(20000))

# reverse the list and return its new head (= 19999).
def work():
    return list(reversed(data))[0]

bench("reverse", work)
